src/framework/MOM_restart.F90

```diff
@@ -460,25 +462,44 @@ subroutine register_restart_field_4d(f_ptr, name, mandatory, CS, longname, units
   character(len=*), optional, intent(in) :: hor_grid  !< variable horizontal staggering, 'h' if absent
   character(len=*), optional, intent(in) :: z_grid    !< variable vertical staggering, 'L' if absent
   character(len=*), optional, intent(in) :: t_grid    !< time description: s, p, or 1, 's' if absent
+  type(axis_info),  dimension(:), &
+                    optional, intent(in) :: extra_axes !< dimensions other than space-time
 
   type(vardesc) :: vd
+  character(len=32), dimension(:), allocatable :: dim_names
+  integer :: n, n_extradims
+
+  if (present(extra_axes)) then
```

Comment:
A comment explaining why `dim_names` is padded would be helpful here.  (My understanding is that it's because `var_desc` was not updated to handle `extra_dims`, which is reasonably outside the scope of this PR.  It should probably be revisited in the future, however)